### PR TITLE
Remove files_filtering and use lookout-sdk-ml

### PR DESCRIPTION
**Pull request by warenlg**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM srcd/lookout-sdk-ml:0.9.0
+FROM srcd/lookout-sdk-ml:0.11.0
```
Comment (vmarkovtsev) [inline Dockerfile]: Oh jeez, this is going to put us into so much trouble. It contains the updated Modelforge which is quite incompatible with all the current models. There is nothing we can do, unfortunately.

File: lookout/style/format/analyzer.py [hunk 2]
```diff
@@ -195,8 +195,11 @@ def check_training_required(
                 _log.warning("language %s is not supported, skipped", language)
                 continue
             overall_lines = changed_lines = 0
-            for file in filter_files(head_files, lang_config["line_length_limit"],
-                                     lang_config["overall_size_limit"], log=_log):
+            for file in parse_files(filepaths=head_files.keys(),
+                                    line_length_limit=lang_config["line_length_limit"],
+                                    overalll_size_limit=lang_config["overall_size_limit"],
```
Comment (vmarkovtsev) [inline lookout/style/format/analyzer.py]: `overalll_size_limit` - too many "l"

File: lookout/style/format/analyzer.py [hunk 3]
```diff
@@ -256,7 +256,7 @@ def train(cls, ptr: ReferencePointer, config: Mapping[str, Any], data_service: D
             else:
                 _log.info("training on %d %s files", len(files), language)
             train_files, test_files = FormatAnalyzer.get_train_test_split(
-                files, lang_config["test_dataset_ratio"], random_state=random_state)
+                files.values(), lang_config["test_dataset_ratio"], random_state=random_state)
```
Comment (vmarkovtsev) [inline lookout/style/format/analyzer.py]: `files` come here directly from `files_by_language` which does not sort them. `get_train_test_split` shuffles without prior sorting, too. We should either make `files_by_language` deterministic or sort in `get_train_test_split` before shuffling. I like the first option more.

**Comment by vmarkovtsev** [comment]
@warenlg Please rebase cleanly, I see many merge commits.

**Comment by vmarkovtsev** [comment]
@zurk Can you please go through the test changes.

File: Dockerfile (hunk 1 above)
Comment (warenlg) [reply Dockerfile]: Yes I saw this this morning, and started to fix some stuff about it

**Comment by zurk** [comment]
Master is fixed. 🎉

**Comment by vmarkovtsev** [comment]
I mean, we need your review @zurk 😃

File: lookout/style/typos/analyzer.py [hunk 4]
```diff
@@ -69,8 +70,13 @@ def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
         base_files_by_lang = files_by_language(c.base for c in changes)
         head_files_by_lang = files_by_language(c.head for c in changes)
         line_length = self.config.get("line_length_limit", self.DEFAULT_LINE_LENGTH_LIMIT)
+        overall_size = self.config.get("overall_size_limit", self.DEFAULT_OVERALL_SIZE_LIMIT)
```
Comment (zurk) [inline lookout/style/typos/analyzer.py]: I think it is worth to have the same structure or config loading as we have in `FormatAnalyser`.
But I see that you just keep the same stricture, so we agree on this we can leave it for @irinakhismatullina :)

**Comment by warenlg** [comment]
Yes @zurk sorry for the big PR, It has been frozen for 1 month due to the paper rush, and I had to fix one month of PRs when porting to the new lookout-sdk-ml.

File: lookout/style/format/tests/test_analyzer.py [hunk 5]
```diff
@@ -73,102 +75,96 @@ def remove_uast(file):
...
-        with lzma.open(str(base / "benchmark.uast.xz")) as fin:
-            cls.uast = bblfsh.Node.FromString(fin.read())
-        cls.base_files = cls.get_files_from_tar(str(base / "freecodecamp-base.tar.xz"))
-        cls.head_files = cls.get_files_from_tar(str(base / "freecodecamp-head.tar.xz"))
         cls.ptr = ReferencePointer("someurl", "someref", "somecommit")
         FeatureExtractor._log.level = logging.DEBUG
         cls.bblfsh_client = bblfsh.BblfshClient("0.0.0.0:9432")
+        parent_loc = Path(__file__).parent.resolve()
+        with tarfile.open(str(parent_loc / "jquery_noisy.tar.xz")) as tar:
+            tar.extractall(path=parent_loc)
```
[the remark this answers is not in the document]
Comment (warenlg) [reply lookout/style/format/tests/test_analyzer.py]: Yes, no need to test on 2 different repos, we can store only one.

File: lookout/style/format/tests/test_analyzer_integration.py [hunk 6]
```diff
@@ -113,14 +113,14 @@ def setUpClass(cls):
         parent = Path(__file__).parent.resolve()
         cls.base_dir_ = tempfile.TemporaryDirectory()
         cls.base_dir = cls.base_dir_.name
-        cls.jquery_dir = os.path.join(cls.base_dir, "jquery")
-        # str() is needed for Python 3.5
         with tarfile.open(str(parent / "jquery.tar.xz")) as tar:
-            tar.extractall(path=cls.base_dir)
+            tar.extractall()
+        # str() is needed for Python 3.5
+        cls.jquery_dir = str(Path(parent) / "jquery")
```
[the remark this answers is not in the document]
Comment (warenlg) [reply lookout/style/format/tests/test_analyzer_integration.py]: yes ofc, but removing it with `shutil.rmtree` is also fine, no ?
Because I need to access those files later in the tests, and with a `TemporaryDirectory`, they disappeared once we get out of the `setUpClass` method.

File: lookout/style/format/utils.py [hunk 7]
```diff
@@ -39,46 +34,6 @@ def GetFiles(self, _) -> Iterable[File]:
         return self.files
 
 
-def prepare_files(filenames: Iterable[str], client: BblfshClient,
```
[the remark this answers is not in the document]
Comment (warenlg) [reply lookout/style/format/utils.py]: yes

File: lookout/style/format/tests/test_analyzer_integration.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (zurk) [reply lookout/style/format/tests/test_analyzer_integration.py]: well, by `Ctrl+C` I mean control-break signal, so, you did not run code in `tearDownClass'.
And if you use the temporary directory the same happens but I believe it will be clean up by the system later.

What about the issue you mentioned try to use
```python
cls.base_dir_ = tempfile.mkdtemp()
```
> The user of mkdtemp() is responsible for deleting the temporary directory and its contents when done with it.

https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp

[the remark this answers is not in the document]
Comment (warenlg) [reply lookout/style/format/tests/test_analyzer_integration.py]: ok got it, let me try that!

**Comment by vmarkovtsev** [comment]
@zurk Can you please continue this: rebase and fix the remaining problems.

**Comment by zurk** [comment]
ok

**Comment by vmarkovtsev** [comment]
@zurk This appears to be tightly coupled with #619 so I will take care of it myself.

**Comment by vmarkovtsev** [comment]
I have looked through the history and see that it is doomed. We need a clean rebase in a new PR. Closing this and waiting for @warenlg to return back next week.

**Comment by zurk** [comment]
Waren, in my experience it is always better to rebase than merge the master branch when you update your codebase. it helps better understand what is really going on in your changes.
